src/uTensor/ops/Arithmetic_kernels.hpp

```diff
@@ -16,5 +16,18 @@ void add_kernel(Tensor& c, const Tensor& a, const Tensor& b) {
     c(i) = static_cast<T>(static_cast<T>(a(i)) + static_cast<T>(b(i)));
 }
 
+template <typename T>
+void mul_kernel(Tensor& c, const Tensor& a, const Tensor& b) {
+  // Decide on c shape
+  TensorShape c_shape = c->get_shape();
+  uint32_t c_size = c_shape.get_linear_size();
+  // TensorInterface& C = reinterpret_cast<TensorInterface*>(*c);
+  // const TensorInterface& A = reinterpret_cast<TensorInterface*>(*a);
+  // const TensorInterface& B = reinterpret_cast<TensorInterface*>(*b);
+
+  for (uint32_t i = 0; i < c_size; i++)
+    c(i) = static_cast<T>(static_cast<T>(a(i)) * static_cast<T>(b(i)));
+}
+
```

Comment:
```suggestion
template <typename T>
void mul_kernel(Tensor& c, const Tensor& a, const Tensor& b) ;

template <>
void mul_kernel<float>(Tensor& c, const Tensor& a, const Tensor& b);

// Dont forget to put definition in CPP file for specialization
template <>
void mul_kernel<float>(Tensor& c, const Tensor& a, const Tensor& b) {
  // Decide on c shape
  TensorShape c_shape = c->get_shape();
  uint32_t c_size = c_shape.get_linear_size();
  // TensorInterface& C = reinterpret_cast<TensorInterface*>(*c);
  // const TensorInterface& A = reinterpret_cast<TensorInterface*>(*a);
  // const TensorInterface& B = reinterpret_cast<TensorInterface*>(*b);

  for (uint32_t i = 0; i < c_size; i++)
    c(i) = static_cast<T>(static_cast<T>(a(i)) * static_cast<T>(b(i)));
}
```

Reply:
FYI the template definition as is doesnt account for quantized forms and doesnt throw any errors that they currently are not supported. I'm ok with merging as is, but I need to go through and mark these fixes in a future PR.